### ISSUE TITLE
Support USB host control IN transfers (IDFGH-6108)

### ISSUE THREAD

**Issue by leemagnusson** [espressif/esp-idf#7786]
I've been working with the new USB Host library on the esp32s2. I'd like to be able to use the control endpoint IN transfers to read data over that endpoint. It seems that your library supports IN transfers well, but the `usb_host_transfer_submit_control` blocks them at the below line.

https://github.com/espressif/esp-idf/blob/1561fbd2c5d001bfe940c3b527d9e94ac549e08c/components/usb/usb_host.c#L1269

changing it to:
```
HOST_CHECK((transfer->bEndpointAddress & 0x7F) == 0, ESP_ERR_INVALID_ARG);
```
gives me a working solution. I do seem to correctly received the callback with IN data from the device.

**Comment by ESP32DE** [comment]
@Dazza0 
yes i know USB2.0 spec - i want say with this, that the "working solution" by @leemagnusson using literal `0x7F` is not good idea, my proposal was,  better using the Mask Var `USB_B_ENDPOINT_ADDRESS_EP_NUM_MASK` cause it is right defined with 0x0F and must(should) not be change.

**Comment by ESP32DE** [comment]
> 
> 
> @leemagnusson Whoops, that check should have been
> 
> ```c
> HOST_CHECK((transfer->bEndpointAddress & USB_B_ENDPOINT_ADDRESS_EP_NUM_MASK) == 0, ESP_ERR_INVALID_ARG); 
> ```
> 
> Will add this fix to the MR along with the #7505 disconnect issue fix.


I can be wrong
shouldn't the check be like this ..:  

HOST_CHECK((transfer->bEndpointAddress & USB_B_ENDPOINT_ADDRESS_EP_NUM_MASK)` !=` 0, ESP_ERR_INVALID_ARG);

**Comment by leemagnusson** [comment]
I just used 0x7F to mean that I bypassed it and didn't want to figure out the correct #define. Also it does work right now with ==. I'm submitting on control transfer endpoint 0. By working solution I mean that I only tested this to do 8 byte in transfers to support the usb dfu protoctol and I am getting valid data back. But I wasn't sure if control in transfers were supported at this time and it was an oversight at this interface or if they were purposely disallowed.

You might also add to the documentation what I figured out experimentally, which was that the transfer size should be `sizeof(usb_setup_packet_t) + num_bytes_in` and that the callback transfer data_buffer will contain the same - that is it will be both the setup packet that is sent out and the data which is read. My first guess was that it would have just returned the `num_bytes_in` portion in the callback.
